src/amuse/community/fi/interface.py

```diff
@@ -44,6 +44,13 @@ class FiInterface(
     MODE_NORMAL = 'normal'
     MODE_NORMAL_OPENMP = 'openmp'
     MODE_PERIODIC_BOUNDARIES   = 'periodic'
+    try:
```

Comment:
its better to put something that does this in the parent class (in its __init__)...you may have to use some trickery, but I am pretty sure its possible...then you don't have to go around changing all interface.py....

Reply:
in CodeInterface or in the LiteratureReferencesMixIn....

Reply:
yes I agree

Reply:
I'm not sure how to get the import path right in that case though...

Reply:
there is examples of this somewhere in AMUSE....but based on:
```inspect.getabsfile(self.__class__)```

Reply:
or maybe getsourcefile...

Reply:
or actually self.__module__ gives you the module...

Reply:
ok I'll check.
In Python 3.8+ it's much easier... Can just do this then
```python
>>> from importlib.metadata import version
>>> version('amuse-fi')
```

Reply:
> or actually `self.__module__` gives you the module...

Ok, I can use that.

Reply:
> ok I'll check.
> In Python 3.8+ it's much easier... Can just do this then
> 
> ```python
> >>> from importlib.metadata import version
> >>> version('amuse-fi')
> ```

you could make it so it uses this when available, and just unknown when not...you still need a way to figure out which module you are though..

Reply:
Fixed it now.